### PR TITLE
1.0/redo auth service local storage

### PR DESCRIPTION
**Pull request by ob6160**
It's not perfect but it will do the job for now.

I've made two preliminary tests, @charlesstrange2 some more need to be made, we should figure out which other scenarios we need to test for.

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -106,3 +112,9 @@ var vm = new Vue({
     // Pass in the router to the Vue instance
     router
 }).$mount('#app'); // Mount the router on the app
+
+EventBus.$on('NetMgr.logout', () => {
+    Store.resetStore();
+
```
Comment (charlesstrange2) [inline src/main.js]: Spacing

File: src/services/netMgr.js [hunk 2]
```diff
@@ -1,17 +1,26 @@
+import Vue from 'vue';
```
Comment (charlesstrange2) [inline src/services/netMgr.js]: Why include this? i don't think we use it in this file.

File: src/services/netMgr.js [hunk 3]
```diff
@@ -35,32 +44,36 @@ var NetMgr = {
...
  *
  * @param tokenData
  */
 NetMgr.setToken = function (tokenData) {
     this.token = tokenData;
+
     if (this.token) {
         this.token.requestTime = Math.floor(Date.now() / 1000);
         this.axiosInstance.defaults.headers.common['Authorization'] = 'Bearer ' + this.token.access_token;
+
```
Comment (charlesstrange2) [inline src/services/netMgr.js:53]: Spacing?

**Review by charlesstrange2 — CHANGES_REQUESTED** [review]
Awesome!
See changes though

File: src/services/netMgr.js (hunk 2 above)
Comment (ob6160) [reply src/services/netMgr.js]: whoops I left that in, good catch

File: src/services/netMgr.js (hunk 3 above)
Comment (ob6160) [reply src/services/netMgr.js:53]: I quite like the space here as those two are variable sets and the last one is a method call. Don't mind either way - your call.

File: src/services/netMgr.js [hunk 4]
```diff
@@ -2,16 +2,24 @@ import Config from '../config.js';
...
-- Authentication
+ NetMgr is a place to put all the
+ - AJAX axios stuff;
+ - Mock binding
+ - Authentication
 
-Though, in theory it could have been split into a series of modules.
+ Though, in theory it could have been split into a series of modules.
+
+ - event bus
```
Comment (katjam) [inline src/services/netMgr.js]: Are these todos? It't really clear what this list is.

File: src/services/netMgr.js (hunk 3 above)
Comment (katjam) [reply src/services/netMgr.js:53]: I am with @charlesstrange2 on this one. Though not much of a leg to stand on - we have been sort of following 'standards' and the rest of the code isn't so gapp-y.

Happy to leave it and wait until we get a standards set agreed.

File: src/services/netMgr.js [hunk 5]
```diff
@@ -160,37 +172,84 @@ NetMgr.mockOff = function () {
...
+    NetMgr.mockOn();
 }
 
+/**
+ * Sets the internal token variable to the result of the NetMgr.getTokenFromLocalStorage function.
+ */
+NetMgr.setTokenFromLocalStorage = function() {
+    let parsedLocalToken = this.getTokenFromLocalStorage();
+
+    if(parsedLocalToken) this.setToken(parsedLocalToken);
```
Comment (katjam) [inline src/services/netMgr.js]: Up to @charlesstrange2 - another standards issue... inline if statement.

**Review by katjam — APPROVED** [review]
@charlesstrange2 I had a few minor comments - but happy for you to merge this in. Looks good.

**Review by charlesstrange2 — APPROVED** [review]

**Pull request merged**